core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -102,9 +114,16 @@ private RetryExecutor(SdkHttpFullRequest request, RequestExecutionContext contex
         }
 
         public CompletableFuture<Response<OutputT>> execute(CompletableFuture<Response<OutputT>> future) throws Exception {
+            MetricRegistry attemptRegistry = MetricUtil.newRegistry(context.executionAttributes());
```

Comment:
Would it make sense to move line 117 - line 120 to `beforeExecute` method

Reply:
Would need to store a reference to the timer then.